src/containers/Topmenu/index.jsx

```diff
@@ -17,28 +18,64 @@ const MenuCont = styled.div`
 class Topmenu extends Component {
     state = {
         categories: [],
+        isLoading: true,
     };
 
+    componentWillMount() {
+        localStorage.getItem('categories') &&
+            this.setState({
+                categories: JSON.parse(localStorage.getItem('categories')),
+                isLoading: false,
+            });
+    }
+
     componentDidMount = () => {
-        this.getCategories();
+        let newCategories = this.getCategories();
+        if (!localStorage.getItem('categories')) {
+            newCategories;
+        }
     };
 
     getCategories = async () => {
         const url = BaseApi.getUrl('categories?_limit=4');
         const response = await fetch(url);
         const categories = await response.json();
-        this.setState(() => ({ categories }));
+        this.setState(() => ({ categories, isLoading: false }));
```

Comment:
O código que você colocou no componentWillUpdate tem alguns problemas.

Primeiro que o método componentWillUpdate está deprecated (https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate).

A outra é que ele está atualizando o localStorage em momentos que não há necessidade, por exemplo, quando vc muda o state no `componentWillMount`, ele vai entrar aqui e vai atualizar o localStorage com o mesmo valor que estava lá.

Minha sugestão é utilizar o callback do `setState`, esse método aceita 2 parâmetros o primeiro é para alterar o `state` de fato, como vc está usando aqui, o segundo é um callback que será chamado logo após o react finalizar a atualização do state (https://reactjs.org/docs/react-component.html#setstate). Ficaria mais ou menos assim:
```suggestion
        this.setState({ categories, isLoading: false }, () => {
                localStorage.setItem('categories', JSON.stringify(categories));
        });
```